sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -389,21 +389,13 @@ public Mono<BlockBlobItem> upload(Flux<ByteBuffer> data, ParallelTransferOptions
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<BlockBlobItem> upload(Flux<ByteBuffer> data, ParallelTransferOptions parallelTransferOptions,
         boolean overwrite) {
-        Mono<Void> overwriteCheck;
-        BlobRequestConditions requestConditions;
-
-        if (overwrite) {
-            overwriteCheck = Mono.empty();
-            requestConditions = null;
-        } else {
-            overwriteCheck = exists().flatMap(exists -> exists
-                ? monoError(LOGGER, new IllegalArgumentException(Constants.BLOB_ALREADY_EXISTS))
-                : Mono.empty());
-            requestConditions = new BlobRequestConditions().setIfNoneMatch(Constants.HeaderConstants.ETAG_WILDCARD);
+        BlobRequestConditions requestConditions = new BlobRequestConditions();
+
+        if (!overwrite) {
+            requestConditions.setIfNoneMatch(Constants.HeaderConstants.ETAG_WILDCARD);
```

Comment:
Is this actually equivalent to the previous implementation? If this ends up using chunked uploading, aka stage blocks and put block list, the ETag validation only applies on put block list. Wouldn't it be better to do a simple exists check before uploading chunks?

Now, this could be moved down the stack to when we know whether upload in full or upload in chunks is being used and if upload in chunks is being used we check for overwriting.